### DISCUSSION TITLE
Is it possible to get whole text of terminal?

### DISCUSSION THREAD

**Discussion by ios37** [migueldeicaza/SwiftTerm#298] [General]
e.g. I have added "ls" it will show all directory the "cd Desktop" I need all raw text.

**Comment by migueldeicaza** [comment]
It depends on how reliable you want this information to be.

You can get access to the current contents displayed as well as what is in the back buffer by accessing the Terminal object and then getting to the active buffer, and from there extract the lines that you want.

But the issue with something like `ls` is that the output might be larger than what the buffer is holding, so this would not work.

An alternative is to replace the print built-in command and have it capture all the output on your behalf.

**Reply by ios37** [reply]
@migueldeicaza Thank you for quick guidance. I have used "script" to log all output in file.

Reference : https://unix.stackexchange.com/questions/200637/save-all-the-terminal-output-to-a-file

**Comment by migueldeicaza** [comment]
If that’s what you want, you already have that information: you are getting it before you even call the terminal’s feed method.

**Reply by ios37** [reply]
Can you please explain more how I can get it before call the terminal's feed?
